pkg/reloader/reloader.go

```diff
@@ -701,7 +708,7 @@ func expandEnv(b []byte) (r []byte, err error) {
 
 		v, ok := os.LookupEnv(string(n))
 		if !ok {
-			err = errors.Errorf("found reference to unset environment variable %q", n)
+			err = errors.Wrapf(expandEnvError, "%s", n)
```

Comment:
IIUC the first failed expansion will stop the processing of the config. I'd rather see this as best-effort (e.g. do all possible replacements and record any error).